Bottom: 7e5877605deb58980ac0a92458f9aa2e0b3d45e5
Top:    1ed7ec1e930018cd3e982543623d32b348d4a15d
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-08 09:19:12 +0800

Refresh of autofs-5.0.7-fix-potential-null-dereference-in-lookup_mount.patch

---

diff --git a/modules/lookup_yp.c b/modules/lookup_yp.c
index 72a8baa..f6095ad 100644
--- a/modules/lookup_yp.c
+++ b/modules/lookup_yp.c
@@ -699,7 +699,7 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 					       mapent, ctxt->parse->context);
 		if (ret) {
 			time_t now = time(NULL);
-			int rv = CHE_FAIL;
+			int rv = CHE_OK;
 
 			cache_writelock(mc);
 			me = cache_lookup_distinct(mc, key);
@@ -707,7 +707,8 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 				rv = cache_update(mc, source, key, NULL, now);
 			if (rv != CHE_FAIL) {
 				me = cache_lookup_distinct(mc, key);
-				me->status = now + ap->negative_timeout;
+				if (me)
+					me->status = now + ap->negative_timeout;
 			}
 			cache_unlock(mc);
 		}
